draft-bradleylundberg-cfrg-arkg.md

```diff
@@ -131,75 +131,39 @@ We expect that additional instances will be defined in the future.
 
+* __Initialization__: The intended private key holder generates a _seed pair_, disclosing the _shared secret seed_ to a subordinate entity while securely retaining the _private seed_.
+* __Public key generation__: With the shared seed, the subordinate entity autonomously generates any number of new public keys alongside a unique _key handle_ for each invocation.
+* __Private key derivation__: The intended private key holder jointly employs the key handle and the private seed to derive the private keys corresponding to the public keys generated by procedure two.
+
+One application is key blinding where the private key is a function of a long-term private key and a freshly generated blinding key, and correspondingly where the public key is a function of the long-term public key and the same blinding key.
 Some motivating use cases of ARKG include:
 
-- Efficient single-use signing keys.
-  The European Union has proposed a digital identity system which, in order to protect users' privacy,
-  needs a unique key pair for each authentication signature.
-  In online usage the system could relatively easily create a key on demand,
-  submit it to a certification authority to have a single-use certificate issued for that key,
-  and then submit that certificate with an authentication signature to a third party to access a service.
-
-  However, the proposed system also includes offline use cases:
-  A user might for example need to use the system in a location with poor or no internet connectivity
-  to present a digital driver's license or authorize a payment.
-  For this, the system may need to pre-emptively generate a large amount of single-use certificates to be used offline.
-
-  One candidate implementation under evaluation to provide signing and key management for this system
-  is the W3C Web Authentication API [WebAuthn] (WebAuthn),
-  which requires a user gesture whenever a WebAuthn operation is invoked.
-  A WebAuthn-based implementation of the proposed digital identity system
-  could use ARKG to pre-emptively generate key pairs for offline use without the need to prompt for a user gesture for each key pair generated.
-
-- Enhanced forward secrecy for encrypted messaging.
-  For example, [section 8.5.4 of RFC 9052][rfc9052-direct-key-agreement] defines COSE representations for encrypted messages and notes that
+- __Single-use asymmetric keys__:
+  Envisioned for the European Union's digital identity framework, ARKG facilitates the generation of unique key pairs for individual authentication processes.
+  Unique single-use asymmetric keys prevent colluding verifiers from using the public key material as a correlation handle.
+  In online usage, ARKG enables public key generation on demand. This is useful when creating single-use certificates, single-use proof of possession keys to be included in (qualified) electronic attestations of attributes or personal identification data attestations.
```

Comment:
The online use case isn't a very strong motivation, as that can almost as easily be done with conventional key generation. The offline use case is much stronger in my opinion: pre-generating keys in large batches at once without having to call out to the secure element (possibly invoking a user gesture) for each key.

Reply:
The online use case for single-use, just-in-time PID/(Q)EAA still looks valid:

- Creating a brand new ECDSA key in the mobile SCE does not provide the issuer any assurance that it is bound to the same device as the PID that was presented to them. With ARKG, the issuer can have this assurance.
- Creating a brand new ECDSA key in the mobile SCE requires additional key management, which brings the latency of key generation and in the Android case also the problem of key deletion. With ARKG, the SCE does not need to do any additional key generation or deletion.